Remove_trivial_connectivity_lines

shireenf-ibm opened the pull request:
#143

adisos commented on pkg/netpol/eval/check.go:
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
 // for connectivity considerations, when requesting allowed connections between 2 workload peers which are the same,
 // looking for 2 different pod instances, if exist (to avoid the trivial case of connectivity from pod to itself)
-func (pe *PolicyEngine) changePodPeerToAnotherPodObject(peer *k8s.PodPeer) {
+// returns true if different pod instances found
```
```suggestion
// if different pod instance was found - returns true and updates the input peer with another pod instance
```

adisos commented on pkg/netpol/eval/check.go:
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
 		dstPodPeer, err := pe.convertWorkloadPeerToPodPeer(dstPeer)
 		if err != nil {
-			return nil, err
+			return nil, isTrivial, err
```
```suggestion
			return nil, false, err
```

adisos commented on pkg/netpol/eval/check.go:
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
 // expecting that srcPeer and dstPeer are in level of workloads (WorkloadPeer)
-func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, error) {
+func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, bool, error) {
+	isTrivial := false
```
```suggestion
```
move the definition of is`Trivial` to be before the call to `changePodPeerToAnotherPodObject` , where it is relevant

adisos commented on pkg/netpol/eval/check.go:
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
+		return conn, isTrivial, err
 	}
-	return nil, fmt.Errorf("cannot have both srcPeer and dstPeer of IP types: src: %s, dst: %s", srcPeer.String(), dstPeer.String())
+	return nil, isTrivial, fmt.Errorf("cannot have both srcPeer and dstPeer of IP types: src: %s, dst: %s", srcPeer.String(), dstPeer.String())
```
```suggestion
	return nil, false, fmt.Errorf("cannot have both srcPeer and dstPeer of IP types: src: %s, dst: %s", srcPeer.String(), dstPeer.String())
```

adisos commented on pkg/netpol/eval/check.go:
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
+// i.e. Connection from deployment to itself when the deployment has only one replica.
 // expecting that srcPeer and dstPeer are in level of workloads (WorkloadPeer)
-func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, error) {
+func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, bool, error) {
```
does it handle identifying trivial connections if the input peers are pods and not workloads? do we have such tests?

shireenf-ibm replied:
seems to me yes it is, 
does the `podlist ` of `tests/ipblockstest` meets the requirement?

adisos reviewed (changes requested):
hi @shireenf-ibm , added a few comments.
The main question is - do we handle both the case of input workload resources, and the case of input pod resources?  please check and see if we have tests for both scenarios.
Also, I tried to check `tests\onlineboutique_with_replicas` and it seems this test contains both pods and deployments resources -- I think it should be split into the two cases separately..

shireenf-ibm commented:
> Also, I tried to check `tests\onlineboutique_with_replicas` and it seems this test contains both pods and deployments resources -- I think it should be split into the two cases separately..

we already had the `tests/onlineboutique_with_replicas_and_variants` which includes only pods resources.
so I removed `pods.yaml` from `tests\onlineboutique_with_replicas`, so it will include deployment resources only .
and added both to the `connlist_test.go ` so we can view their connections list.

for `tests\onlineboutique_with_replicas` , since it doesn't include pods resources any more, I had to remove its `tests/onlineboutique_with_replicas/connections_map_with_replicas_output.txt` and eliminate the relevant test in `eval_test.go`, as it creats an empty `podsMap` now.
